controller/node_controller.go

```diff
@@ -995,6 +1010,199 @@ func (nc *NodeController) syncNodeStatus(pod *corev1.Pod, node *longhorn.Node) e
 	return nil
 }
 
+func (nc *NodeController) environmentCheck(kubeNode *corev1.Node, node *longhorn.Node) {
+	nc.syncKernelVersion(kubeNode, node)
+	namespaces := []lhtypes.Namespace{lhtypes.NamespaceMnt, lhtypes.NamespaceNet}
+	nc.syncPackagesInstalled(kubeNode, node, namespaces)
+	nc.syncMultipathd(node, namespaces)
+	nc.syncNFSClientVersion(kubeNode, node, namespaces)
+}
+
+func (nc *NodeController) syncKernelVersion(kubeNode *corev1.Node, node *longhorn.Node) {
+	kernelVersion := kubeNode.Status.NodeInfo.KernelVersion
+	brokenKernelVersion := []string{"5.15.0", "6.5.6"}
+	majorKernel5150Version := 5
+	majorKernel656Version := 6
+	brokenKernel5150PatchVersion := 94
+	brokenKernel656MinorVersion := 6
+	fixedKernel5150PatchVersion := 100
+
+	for _, version := range brokenKernelVersion {
+		if !strings.Contains(kernelVersion, version) {
+			continue
+		}
+		kVersion, err := parseKernelVersion(kernelVersion)
+		if err != nil {
+			node.Status.Conditions = types.SetCondition(node.Status.Conditions, longhorn.NodeConditionTypeSupportedKernel, longhorn.ConditionStatusFalse,
+				string(longhorn.NodeConditionReasonKernelNotSupport),
+				fmt.Sprintf("Failed to parse kernel version %v: %v", kernelVersion, err))
+			return
+		}
```

Comment:
Move the parsing logic outside of the for loop.
It only needs only be parsed once.

Reply:
Removed the kernel check.